Feature/mypage/usermodify

DevPCW opened the pull request:
마이페이지
[유저 회원정보 수정]
1. 닉네임 수정 api 개발
2. 이미지 수정 api 개발

Issue: #7

DevPCW committed:
[feat] 마이페이지 회원 정보 수정 - 닉네임 수정 / 이미지 수정 api 개발 초안

DevPCW committed:
[feat] 마이페이지 회원 정보 수정 - 닉네임 수정 / 이미지 수정 api 개발 초안

DevPCW committed:
[refactor] : 불필요한 이미지 파일 삭제 #7

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/SurveyManagementBackendApplication.java:
```diff
@@ -8,6 +8,8 @@ public class SurveyManagementBackendApplication {
 		SpringApplication.run(SurveyManagementBackendApplication.class, args);
+
+		System.out.println("hello");
 	}
```
main 메서드에 코드는 불필요해보입니다.
또한 그 코드가 System.out.println("hello"); 라면 더욱 불필요 해보입니다.
삭제하는게 어떨까요?

kl204 replied:
이하동문

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/common/utils/MultipartFileTypeHandler.java:
```diff
@@ -0,0 +1,45 @@
+        // Implement the conversion from byte[] to MultipartFile if needed.
+        return null;
+    }
+}
```
해당 코드는 이제 사용하지 않는다고 말씀해주신것 같습니다.
사용하지 않는 코드는 삭제해도 무방할것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/common/utils/WebMvcConfig.java:
```diff
@@ -0,0 +1,39 @@
+//                    // 디코딩을 해서 다시 원래 자원 경로에 잡아줌.
+//                });
+//    }
+}
```
사용하지 코드의 주석은 삭제해주시면 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/controller/UserApi.java:
```diff
@@ -0,0 +1,72 @@
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(CommonResponse.error(ErrorResponse.of("Modify Fail")));
+        }
+
```
int형을 받아서 처리하는것 보단 boolean 반환이 더 괜찮을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/controller/UserApi.java:
```diff
@@ -0,0 +1,72 @@
+
+
+    @PatchMapping("/modify-nickname")
+    public ResponseEntity<CommonResponse> userNickNameModify(@RequestBody UserModifyDTO userModifyDTO) {
```
RESTful API 규칙을 찾아봐서 API path를 조금 수정해주시면 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/controller/UserApi.java:
```diff
@@ -0,0 +1,72 @@
+
+
+
+
```
`//` 형식의 주석보다는 저희는 `javaDoc` 을 사용하기로 컨베션을 지정했었습니다.
`javaDoc`을 이용한 주석을 메서드와 클래스단에 달아 주시는게 더욱 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/domain/User.java:
```diff
@@ -0,0 +1,27 @@
+    private String userEmail;
+    private String userImage;
+    private LocalDateTime createdAt;
+}
```
도멘인 클래스의 경우에는 final 키워드를 붙여주시는것이 좋아보입니다.

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/dto/request/ImageUploadDTO.java:
```diff
@@ -0,0 +1,16 @@
+    private Long userNo;
+//    private MultipartFile imageFile;
+    private byte[] imageData; // Add this variable to hold image data
+}
```
이 부분도 주석으로 처리하기보다는 다른 방법을 사용하시는게 더 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/dto/request/UserDTO.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.util.Date;
+
+@Data
```
`@Data` 애너테이션에서는 `@Setter` 애너테이션이 포함되어 있습니다.
무분별한 Setter의 경우 데이터가 변경될 수 있으므로 필요한 애너테이션만을 선택해서 사용하는게 좋을것 같습니다!

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/dto/request/UserModifyDTO.java:
```diff
@@ -0,0 +1,18 @@
+@Data
+public class UserModifyDTO {
+    private Long userNo;
+
```
Mybatis의 경우에는 래퍼 타입보다는 원시 타입을 더 권장하는것으로 알고 있습니다.
이 부분 한번 참고해보시면 어떨까요?

MyeoungDev commented on src/main/java/com/douzone/surveymanagement/user/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,59 @@
+        } else {
+            System.out.println("Failed to modify userNickName.");
+        }
+
```
`System.out.println` 의 경우에는 정말 필요한 경우 `log` 를 사용해주시면 더 좋을 것 같습니다.
이 또한 로그 레벨을 고려해서 사용해주시고,  해당 코드 같이 확인과 디버깅을 위한 로그는 삭제해주셔도 좋을것 같습니다

MyeoungDev commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,33 @@
+spring.servlet.multipart.max-request-size: 10MB
+spring.servlet.multipart.resolve-lazily: true
+spring.servlet.multipart.file-size-threshold: 0
+spring.servlet.multipart.location: /tmp
```
yml 같은 설정파일은 `.gitignore` 를 통해서 안올라오게 해주시면 좋을것 같습니다.
제가 설정할때 `.properties` 파일 같은 경우에는 gitignore에 추가하였는데 yml은 빠진것 같습니다

MyeoungDev commented on src/main/resources/mybatis-config/mybatis-config.xml:
```diff
@@ -0,0 +1,9 @@
+    <typeHandlers>
+        <typeHandler handler="com.douzone.surveymanagement.common.utils.MultipartFileTypeHandler"/>
+    </typeHandlers>
+</configuration>
```
이것또한 앞서 이야기했던 사용되지 않는 config 파일인것으로 알고있습니다.
만약 사용되지 않는 파일의 경우 삭제해주셔도 좋을것 같습니다.

MyeoungDev reviewed (commented):
고생하셨습니다.
코멘트 단 부분 확인하시고 다시 부탁드립니다!

kl204 commented on src/main/java/com/douzone/surveymanagement/user/service/impl/FileUploadServiceImpl.java:
```diff
@@ -0,0 +1,73 @@
+    @Override
+    @Transactional
+    public String uploadImageAndUpdateUser(Long userNo, MultipartFile file) throws IOException {
+        // 파일을 업로드하고 저장된 파일의 경로를 가져옴
```
// 주석의 경우 Discussion의 Code Convention 참고하여 작성하면 더 좋을듯!

kl204 reviewed (approved):
주석 처리 정리하는 부분 외에는 굳잡

DevPCW committed:
[refactor] : 1. main 메서드의 불 필요한 코드 삭제
         2. 사용하지 않는 코드 삭제 / 불필요한 주석 삭제
         3. RESTful API 규칙 적용(mapping url 부분)
         4. 필요한 부분에 javaDoc 추가
         5. User 도메인 클래스 누락 코드 추가
         6. @Data 어노테이션 제거 -> @Getter 사용(Setter 메소드 사용한 부분 리펙토링)
         7. Long 타입 -> long으로 수정(래퍼타입 -> 원시타입)
         8. 이전 사용자 이미지를 지정 이미지 경로 폴더에서 삭제하는 기능을 이미지 수정 서비스 로직에 추가
         9. PatchMapping을 사용하였던 api들을 PutMapping으로 수정
         10. 기존 이름과 역할이 매칭이 되지 않는 메소드들의 메소드명을 알아보기 쉽게 직관적으로 수정 #7

MyeoungDev reviewed (approved):
고생하셨습니다 👍
